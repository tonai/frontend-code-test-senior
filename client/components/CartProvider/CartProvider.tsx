import { ReactNode, useMemo, useState } from "react";

import { cartContext } from "../../contexts/cart";
import { ICart } from "../../types/cart";

export interface ICartProviderProps {
  children: ReactNode;
}

export default function CartProvider(props: ICartProviderProps) {
  const { children } = props;
  const [cart, setCart] = useState<ICart>({ products: {} });
  const contextValue = useMemo(
    () => ({
      cart,
      setCart,
    }),
    [cart]
  );

  return (
    <cartContext.Provider value={contextValue}>{children}</cartContext.Provider>
  );
}
