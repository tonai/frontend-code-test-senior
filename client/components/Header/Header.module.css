.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 17px 0 28px;
}

.cart {
  position: relative;
}

.quantity {
  position: absolute;
  top: -10px;
  right: -10px;
  width: 20px;
  height: 20px;
  background-color: var(--sohoLights);
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
}