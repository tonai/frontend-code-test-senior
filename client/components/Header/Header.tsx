import Image from "next/image";

import css from "./Header.module.css";

export default function Header() {
  return (
    <header className={css.header}>
      <Image
        src="/octopus-logo.svg"
        alt="Octopus Energy Logo"
        width={192}
        height={27}
      />
      <Image src="/basket.svg" alt="Basket" width={30} height={27} />
    </header>
  );
}
