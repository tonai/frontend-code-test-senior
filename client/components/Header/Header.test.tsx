import { render } from "@testing-library/react";

import Header from "./Header";
import CartProvider from "../CartProvider/CartProvider";

describe("Header component", () => {
  it("matches snapshot", () => {
    const { container } = render(<Header />, {
      wrapper: CartProvider,
    });
    expect(container).toMatchSnapshot();
  });
});
