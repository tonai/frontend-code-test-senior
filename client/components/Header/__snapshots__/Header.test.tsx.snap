// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <img
      alt="Octopus Energy Logo"
      data-nimg="1"
      decoding="async"
      height="27"
      loading="lazy"
      src="/octopus-logo.svg"
      style="color: transparent;"
      width="192"
    />
    <div
      class="cart"
    >
      <img
        alt="Basket"
        data-nimg="1"
        decoding="async"
        height="27"
        loading="lazy"
        src="/basket.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
  </header>
</div>
`;
